src/factory.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Factory shared by GitHub Action and CLI for creating Release PRs
+// and GitHub Releases:
+
+import {ReleasePR, ReleasePROptions} from './release-pr';
+import {RubyReleasePROptions} from './releasers/ruby';
+import {
+  GitHubReleaseOptions,
+  GitHubRelease,
+  ReleaseResponse,
+} from './github-release';
+import {getReleasers} from './releasers';
+
+export function runCommand(
```

Comment:
Do you need these extra exports if they are included into the single `factory` export?

Reply:
Is this extra factory necessary? The only place this is called is in the argv parsing and at that point you already know which command you're trying to execute.

Reply:
I was thinking it might be nice in the action, but maybe it's overkill.

Reply:
@chingor13 actually, one reason I would like to keep this approach, was that it made it really nice to test the CLI:

```js
      let classToRun: GitHubRelease;
      // This logic is used to capture the class that would
      // be executed, allowing us to validate that the appropriate
      // properties were assigned from the argument parser:
      sandbox.replace(
        factory,
        'run',
        (runnable): Promise<undefined> => {
          classToRun = runnable as GitHubRelease;
          return Promise.resolve(undefined);
        }
      );
      parser.parse(
        'github-release --repo-url=googleapis/release-please-cli --package-name=cli-package'
      );
```

Because I was able to mock just the logic that runs the GitHub Release or Release PR, without actually running it.